### ISSUE TITLE
Automate hunt feature turns on Status (Standout) for some units

### ISSUE THREAD

**Issue by Thieffi** [caveman2cosmos/Caveman2Cosmos#885]
I tried to lessen micromanaging by leveling up my hunters' camouflage and send them to hunt hoping they'd be hidden from barbarians. Occasionally a barbarian unit killed a hunter so I thought my unit's camouflage value wasn't high enough, but then I realized that some of my hunters had enabled the Standout status making their stealth levels meaningless.

I tested this further by building three Ambushers. Ambusher 2 functions well with Automate Hunt and stays hidden while going around killing stuff. Ambushers 1 and 3 however turn on Standout the turn after I click the Automate hunt button, and if I click the button to remove Standout status while they're automated, they will turn it back on. The units are identical to each others, so I don't know why two of them don't like stealth.

**Comment by Thunderbrd** [comment]
Although perhaps the coding could be more specific, this happens so that hunters and other stealth units can protect the animals they need to escort, and other situations where they are trying to protect something they're escorting or have captured.  While it could be refined, I figured I would at least say how and what triggers the standout usage.

**Comment by raxo2222** [comment]
> 
> 
> Although perhaps the coding could be more specific, this happens so that hunters and other stealth units can protect the animals they need to escort, and other situations where they are trying to protect something they're escorting or have captured. While it could be refined, I figured I would at least say how and what triggers the standout usage.

This should be disabled if playing with "teleport hunting rewards" game option.
That is it shouldn't be triggered, if subdued animals are simply teleported.

**Comment by Thieffi** [comment]
As it is, it makes it meaningless to give stealth-upgrades to a unit that you want to automate or automating stealthy units. It's also unclear why some units of the same type stay stealthy while others embrace the spirit of Leroy Jenkins. It might work better, if the automation kept the Standout status set by the player; that way it would be possible to turn it on when needed to escort a subdued animal or a captive.

**Comment by Thunderbrd** [comment]
It's also to protect the generals that get sent out with hunting parties.  It seems easy enough to say turn it off if teleport hunting rewards is on but there's captives and workers that get subdued as well.  It's supposed to at least think it knows what it's doing but the actual coding would have to be fully studied to see how to improve the logic.
Not saying this is a needless report - obviously it is, just saying it's a little complicated.  I thought it used to be working more appropriately so I wonder if its been tweaked since I set it up?  Meh, I was rushed as I was putting it together anyhow and figured it would need further refining eventually.  AI on statuses is a bitch.

**Comment by Thieffi** [comment]
For now it seems the World Builder can be used as a workaround. If there's a stealth unit I want to automate but who goes to Standout mode, I open the World Builder, create a new unit with the same upgrades and XP, and delete the old one.

**Comment by Thunderbrd** [comment]
I tend to see slightly poor automation as a feature rather than a bug anyhow - best to force the player to have to manually play the units than to let them lazy it out and let the game play itself.

**Comment by raxo2222** [comment]
> 
> 
> I tend to see slightly poor automation as a feature rather than a bug anyhow - best to force the player to have to manually play the units than to let them lazy it out and let the game play itself.
> 
> That said, it does reveal where the AI is doing some less than optimal stuff.

Isn't this stuff, that AI uses normally?

**Comment by Thunderbrd** [comment]
Yeah.  So it could be a little improved but it would take figuring out where the logic is flawed.

**Comment by raxo2222** [comment]
So for example AI doesn't know when to start and stop using statuses effectively.

**Comment by Thunderbrd** [comment]
There are flaws but it does have a sort of logic it's using.  I'm not sure what his example actually is since I can't see his game but if one was to read the coding and see why it's thinking it should be standing out, it might help.  It's possible it went to standout for good cause and just hasn't yet found the cause yet to go back invisible since it takes an action to make the switch.

**Comment by raxo2222** [comment]
"No longer escorting unit" should be good enough action to remove "stand out" status, if originally it was without it ;)

**Comment by Thunderbrd** [comment]
yeah if they have time - the thing is 'escorting' isn't a status that's tracked so you have to evaluate it in the code.

**Comment by Thunderbrd** [comment]
If that's the case, I assume it must be that the default AI type on those units may be the problem.  That or it makes the determination based on what is on the plot in the city they start in.
